pipeline/src/aggregator/mod.rs

```diff
@@ -802,38 +802,47 @@ impl Aggregator {
             .await
             .context("writing to mem table data")?;
 
-        let count = self
-            .ctx
-            .table(EVENT_STATES_MEM_TABLE)
-            .await?
-            .count()
-            .await
-            .context("count mem table")?;
+        let count = self.count_cache().await?;
         // If we have enough data cached in memory write it out to persistent store
         if count >= self.max_cached_rows {
-            self.ctx
-                .table(EVENT_STATES_MEM_TABLE)
-                .await?
-                .write_table(
-                    EVENT_STATES_PERSISTENT_TABLE,
-                    DataFrameWriteOptions::new()
-                        .with_partition_by(vec!["event_cid_partition".to_owned()]),
-                )
-                .await
-                .context("writing to persistent table")?;
-            // Clear all data in the memory batch, by writing an empty batch
-            self.ctx
-                .read_batch(RecordBatch::new_empty(schemas::event_states_partitioned()))
-                .context("reading empty batch")?
-                .write_table(
-                    EVENT_STATES_MEM_TABLE,
-                    DataFrameWriteOptions::new().with_insert_operation(InsertOp::Overwrite),
-                )
-                .await
-                .context("clearing mem table")?;
+            self.flush_cache().await?;
         }
         ordered.collect().await.context("collecting ordered events")
     }
+
+    async fn count_cache(&self) -> Result<usize> {
+        self.ctx
+            .table(EVENT_STATES_MEM_TABLE)
+            .await?
+            .count()
+            .await
+            .context("count mem table")
+    }
+
+    async fn flush_cache(&self) -> Result<usize> {
+        let cnt = self.count_cache().await?;
```

Comment:
we double count in the base case (check count, if over our threshold, flush where we count again for logging purposes).. it's probably not a bit deal, but pointing it out